GrandCentralBoard/Widgets/Watch/View/WatchWidgetView.swift

```diff
@@ -46,11 +47,14 @@ final class WatchWidgetView: UIView, ViewModelRendering {
 
     private func handleTransitionFromState(state: RenderingState<ViewModel>?, toState: RenderingState<ViewModel>) {
         switch (state, toState) {
-            case (_, .Rendering(let viewModel)):
-                setUpLabelsWithViewModel(viewModel)
-                setUpImagesWithViewModel(viewModel)
-            default:
-                break
+        case (_, .Rendering(let viewModel)):
+            removeFailureView()
+            setUpLabelsWithViewModel(viewModel)
+            setUpImagesWithViewModel(viewModel)
+        case (_, .Failed):
+            showFailureView()
```

Comment:
If widget is failing for a long time `fillViewWithView` can be called multiple times. It leads to situation where `failureView` is added multiple times as a subview.